models/git_diff.go

```diff
@@ -418,6 +418,44 @@ func GetDiffRange(repoPath, beforeCommitID string, afterCommitID string, maxLine
 	return diff, nil
 }
 
+func GetRawDiff(repoPath, commitId string, diffType string) (string, error) {
+	repo, err := git.OpenRepository(repoPath)
+	if err != nil {
+		return "", err
+	}
+
+	commit, err := repo.GetCommit(commitId)
+	if err != nil {
+		return "", err
+	}
+
+	var cmd *exec.Cmd
+	if diffType == "diff" {
```

Comment:
We should use `select diffType {}` and return error is we do not recognize that type.